src/Orleans.Core/Core/ClusterClient.cs

```diff
@@ -14,6 +17,7 @@ namespace Orleans
     internal class ClusterClient : IInternalClusterClient
     {
         private readonly OutsideRuntimeClient runtimeClient;
+        private readonly ClusterClientLifecycle clusterClientLifecycle;
```

Comment:
use `IClusterClientLifecycle` here? instead of the concrete class

Reply:
The IClusterClientLifecycle interface is observable portion only, used by application layer to register with the lifecycle.  This logic uses both the observable and observer calls, so full impl is used.  For further abstraction we can store two interfaces (like we do with the grain lifecycle), but I'm unclear how valuable that is here.  It's done that way for the grain lifecycle because there is separate logic using the separate interfaces.